### ISSUE TITLE
Adding projects from submodules leads to some incompatibilities with plugins

### ISSUE THREAD

**Issue by leandrolerena** [squidfunk/mkdocs-material#6970]
### Context

I use the projects plugin to evaluate mkdocs for a documentation project. Therefore, I need documentation to be aggregated from different repositories. 

In particular, I have two issues, one with git-authors, the other with pymdownx.snippets, which make me wonder if the following statement:

> Since the projects plugin will invoke local builds in the directory of where your submodules are located, this plugin should work without problems, as it does not use temporary directories.

as the local build works perfectly.

See discussion on git-authors here: https://github.com/timvink/mkdocs-git-authors-plugin/issues/79

### Bug description

git-authors plugin reports the following on the project included:
```
WARNING -  [project://.zebbra-mkdocs-dummyproject] [git-authors-plugin] /home/USER/projects/zebbra-mkdocs/submodules/zebbra-mkdocs-dummyproject/docs/index.md has not
           been committed yet. Lines are not counted
...
```

pymdownx.snippets do not work 

### Related links

- https://github.com/timvink/mkdocs-git-authors-plugin/issues/79

### Reproduction

[reproduction.zip](https://github.com/squidfunk/mkdocs-material/files/14742147/reproduction.zip)

### Steps to reproduce

1. Clone https://github.com/zebbra/zebbra-mkdocs/tree/mkdocs_projects (make sure the branch is mkdocs_projects, as I am testing different approaches)
2. git submodule init
3. git submodule update
4. pipenv install
5. make serve

This will reproduce the error from snippets, you can remove the snippets by clearing the file content in `submodules/zebbra-mkdocs-dummyproject/docs/index.md`

6. Open in browser
7. Click on Test in the navigation (this will lead to the project included with the plugin)
8. Observe that git authors are empty)

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting. @alexvoss I'm pretty sure this is related to the cwd issue you encountered. We'll look into it, but please understand that it might take some time – it is tricky to fix and will require us to rewrite some things, but I agree that this should definitely work.

**Comment by squidfunk** [comment]
Thanks for reporting. Fixed in [`f2f919c`](https://github.com/squidfunk/mkdocs-material-insiders/commit/f2f919cebed30e24ae5db5ea578e1038c775b6d3). The projects plugin will now set the current working directory to the directory the `mkdocs.yml` is located in and build the project from there. This should resolve all known issues with relative paths used in plugins and extensions when using the projects plugin.

There's a problem with your reproduction however that makes it crash regardless whether you're using the projects plugin or not. Snippets are stored inside `docs`, but the path is incorrect in `index.md`:

```` md
...
Reference the whole python file 

```python
--8<-- "dummyproject-snippets/local.py"      # <- incorrect 
--8<-- "docs/dummyproject-snippets/local.py" # <- correct
```
...
````

After prepending `docs/` to both `local.py` snippet URLs, the build succeeds with and without the plugin.

**Comment by squidfunk** [comment]
Released as part of [9.5.17+insiders-4.53.6](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.53.6).

**Comment by leandrolerena** [comment]
Thanks @squidfunk , that was a lot faster than anticipated! 

And yes, even though the reproduction works on my side, it contains a nasty hack to circumvent an issue with paths for snippets (duplicate paths) which you just fixed. 

I'll be happy to test everything in a clean environment. Have a good start into the week!
